plugins/dynamix.docker.manager/include/DockerClient.php

```diff
@@ -354,14 +355,47 @@ public function getIcon($Repository,$contName,$tmpIconUrl='') {
 			@copy($icon, $iconRAM);
 		}
 		if (!is_file($icon) && is_file($iconRAM)) {
-			@copy($iconRAM,$icon);
+			@copy($iconRAM, $icon);
 		}
 		if ( !is_file($iconRAM) ) {
-			exec("/usr/bin/logger ".escapeshellarg("$imageName: Could not download icon $imgUrl"));
+			exec("/usr/bin/logger ".escapeshellarg("$contName: Could not download icon $imgUrl"));
+		}
+		else {
+			$this->purgeUnusedIconFiles($contName, $iconFile);
 		}
 
 		return (is_file($iconRAM)) ? str_replace($docroot, '', $iconRAM) : '';
 	}
+
+	public function purgeUnusedIconFiles($contName, $keepIcon='') {
+		global $docroot, $dockerManPaths;
+
+		$icon_glob = sprintf('%s/%s-*.png', $dockerManPaths['images-ram'], $contName);
+		$ramFiles = glob($icon_glob);
+		foreach ($ramFiles as $filename) {
+			if ( ($keepIcon === '') || !(strpos($filename, $keepIcon) !== false) ) {
+				@unlink($filename);
+			}
+		}
```

Comment:
The first block removes all icons associated with contName in images-ram except the one specified nu keepIcon.